Objects/longobject.c

```diff
@@ -32,7 +32,9 @@ static inline stwodigits
 medium_value(PyLongObject *x)
 {
     assert(IS_MEDIUM_VALUE(x));
-    return ((stwodigits)Py_SIZE(x)) * x->ob_digit[0];
+    Py_ssize_t s = Py_SIZE(x);
+    stwodigits d = x->ob_digit[0];
+    return s == 1 ? d : s == -1 ? -d : 0;
```

Comment:
I have thought this could avoid `MUL` instruction, but the benchmark proves this change brings no better or worse performance. I'll drop it.

Reply:
From what I understand, on most modern CPUs, a multiplication isn't *that* much more expensive than (say) addition, but branches like this can be quite expensive if they are unpredictable.

Reply:
Indeed, the multiplication spelling was deliberately introduced recently in a PR targeting speed of integer operations. See #27832. Please revert this change.

Reply:
@xuxinhang This change still needs to be reverted - it's unrelated to the main purpose of the PR.